game-core/src/main/java/games/strategy/engine/data/NamedUnitHolder.java

```diff
@@ -1,4 +1,7 @@
 package games.strategy.engine.data;
 
+/**
+ * A named object that contains a collection of {@link Unit}s.
+ */
 public interface NamedUnitHolder extends Named, UnitHolder {
```

Comment:
This is a bit curious, any consideration to fix this one by inlining? (eg: `implements Named, UnitHolder`)

Reply:
Don't think it's possible without other refactoring.  `UnitCollection#getHolder()` returns a reference to this type.

Reply:
Shall we deprecate this then? That way we would capture some useful results from this conversation and capture our observations after inspecting this code. I'm pretty sure we'll eventually forget, or someone else may not see, that this class should not be used in favor of direct usage of the composite interfaces

Reply:
Since `Player` and `Territory` are the only types that implement `UnitHolder`, and they both implement it through `NamedUnitHolder`, it might be easier to just add a `getName()` method to `UnitHolder` and get rid of `NamedUnitHolder`.  I think that's backwards compatible because  information about implemented interfaces is not serialized.  I will investigate further and submit a follow-up PR if it's possible.

Reply:
I played around with removing `NamedUnitHolder`, as discussed above.  Below are the changes I made:

```diff
diff --git a/game-core/src/main/java/games/strategy/engine/data/NamedUnitHolder.java b/game-core/src/main/java/games/strategy/engine/data/NamedUnitHolder.java
deleted file mode 100644
index 887217b54..000000000
--- a/game-core/src/main/java/games/strategy/engine/data/NamedUnitHolder.java
+++ /dev/null
@@ -1,7 +0,0 @@
-package games.strategy.engine.data;
-
-/**
- * A named object that contains a collection of {@link Unit}s.
- */
-public interface NamedUnitHolder extends Named, UnitHolder {
-}
diff --git a/game-core/src/main/java/games/strategy/engine/data/PlayerID.java b/game-core/src/main/java/games/strategy/engine/data/PlayerID.java
index 32ed2cae3..8c4043964 100644
--- a/game-core/src/main/java/games/strategy/engine/data/PlayerID.java
+++ b/game-core/src/main/java/games/strategy/engine/data/PlayerID.java
@@ -11,7 +11,7 @@ import games.strategy.triplea.delegate.Matches;
 /**
  * A game player (nation, power, etc.).
  */
-public class PlayerID extends NamedAttachable implements NamedUnitHolder {
+public class PlayerID extends NamedAttachable implements UnitHolder {
   private static final long serialVersionUID = -2284878450555315947L;
 
   private static final String DEFAULT_TYPE_AI = "AI";
diff --git a/game-core/src/main/java/games/strategy/engine/data/Territory.java b/game-core/src/main/java/games/strategy/engine/data/Territory.java
index a056c4ce5..207b8f4aa 100644
--- a/game-core/src/main/java/games/strategy/engine/data/Territory.java
+++ b/game-core/src/main/java/games/strategy/engine/data/Territory.java
@@ -7,7 +7,7 @@ import javax.annotation.Nullable;
 /**
  * A territory on a map.
  */
-public class Territory extends NamedAttachable implements NamedUnitHolder, Comparable<Territory> {
+public class Territory extends NamedAttachable implements UnitHolder, Comparable<Territory> {
   private static final long serialVersionUID = -6390555051736721082L;
   private final boolean m_water;
   private PlayerID m_owner = PlayerID.NULL_PLAYERID;
diff --git a/game-core/src/main/java/games/strategy/engine/data/UnitCollection.java b/game-core/src/main/java/games/strategy/engine/data/UnitCollection.java
index 83ec66576..ba6dc9117 100644
--- a/game-core/src/main/java/games/strategy/engine/data/UnitCollection.java
+++ b/game-core/src/main/java/games/strategy/engine/data/UnitCollection.java
@@ -19,7 +19,7 @@ import games.strategy.util.IntegerMap;
 public class UnitCollection extends GameDataComponent implements Collection<Unit> {
   private static final long serialVersionUID = -3534037864426122864L;
   private final List<Unit> m_units = new ArrayList<>();
-  private final NamedUnitHolder m_holder;
+  private final UnitHolder m_holder;
 
   /**
    * Creates new UnitCollection.
@@ -27,7 +27,7 @@ public class UnitCollection extends GameDataComponent implements Collection<Unit
    * @param holder named unit holder
    * @param data game data
    */
-  public UnitCollection(final NamedUnitHolder holder, final GameData data) {
+  public UnitCollection(final UnitHolder holder, final GameData data) {
     super(data);
     m_holder = holder;
   }
@@ -176,7 +176,7 @@ public class UnitCollection extends GameDataComponent implements Collection<Unit
     return getPlayersWithUnits().size() > 1;
   }
 
-  public NamedUnitHolder getHolder() {
+  public UnitHolder getHolder() {
     return m_holder;
   }
 
diff --git a/game-core/src/main/java/games/strategy/engine/data/UnitHolder.java b/game-core/src/main/java/games/strategy/engine/data/UnitHolder.java
index eb7dc48df..f195c93e1 100644
--- a/game-core/src/main/java/games/strategy/engine/data/UnitHolder.java
+++ b/game-core/src/main/java/games/strategy/engine/data/UnitHolder.java
@@ -3,7 +3,7 @@ package games.strategy.engine.data;
 /**
  * An object that contains a collection of {@link Unit}s.
  */
-public interface UnitHolder {
+public interface UnitHolder extends Named {
   String TERRITORY = "T";
   String PLAYER = "P";
 
```

With this change, I was able to load some old save games and play a network game with a 9687 client.

However, while grepping an old save game, I found a reference to `NamedUnitHolder` in the class descriptor for `UnitCollection` (in the `m_holder` field descriptor).  Therefore, I'm not sure why an exception is not raised when deserializing `UnitCollection` instances.  (It may be that the field descriptors aren't resolved by the default deserialization process, and would only be required when, for example, overriding `ObjectInputStream#resolveClass()`.)  So, even though the tests above seem to indicate getting rid of `NamedUnitHolder` works, I don't feel comfortable following through on that given that a reference to this type exists in save games in the wild.